pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -2170,6 +2182,10 @@ public void setFetchSize(int rows) throws SQLException {
       throw new PSQLException(GT.tr("Fetch size must be a value greater to or equal to 0."),
           PSQLState.INVALID_PARAMETER_VALUE);
     }
+    // ignore request to set fetchSize to 0 if it is currently > 0
+    if (fetchSize > 0 && rows == 0) {
+      return;
+    }
```

Comment:
Could you please clarify why ignore the new fetch size?
I guess users might want to adjust fetchSize as they fetch, so it might be unexpected to ignore the new values.

Reply:
If you set the fetchSize to 0 here we will stop fetching more data.

Reply:
We should treat `fetchSize` like a hint rather than an "exact value to use for fetching". Then `0` would not cause "0 rows to fetch"

Reply:
The initial value for `fetchSize` is `0`, so ignoring `setFetchSize(0)` sounds arbitrary

Reply:
fetch size of zero is specifically called out as special `If the fetch size specified is zero, the JDBC driver ignores the value and is free to make its own best guess as to what the fetch size should be.`